### DISCUSSION TITLE
ldt compile error-undefined reference to symbol 'SZ_BufftoBuffCompress

### DISCUSSION THREAD

**Discussion by jachungithub** [NASA-LIS/LISF#1032] [Q&A] [answered]
Hi,

New error messages were encountered  while LDT ***(LISF_public_release_7.3.3)*** was being compiled.. 
FYI, the Intel Comiler version is 21.3.
Any suggestions would be really appreciated.
Thank you,

> ***error messages***

ld: /opt/kma/kma_lib/apps/hdf4/4.2.15_fort/INTEL/210/lib//libdf.a(cszip.o): undefined reference to symbol 'SZ_BufftoBuffCompress'
//opt/kma/kma_lib/apps/szip/2.1.1/lib/libsz.so.2: error adding symbols: DSO missing from command line
make: *** [LDT] Error 1

[ERR] Compile failed

> ***configure.ldt***

FC              = mpiifort
FC77            = mpiifort
LD              = mpiifort
CC              = mpiicc
AR              = ar
MOD_ESMF        = /opt/kma/kma_lib/apps/esmf_mpi/710R/INTEL/210/mod/modO/Linux.intel.64.intelmpi.default
LIB_ESMF        = /opt/kma/kma_lib/apps/esmf_mpi/710R/INTEL/210/lib/libO/Linux.intel.64.intelmpi.default
INC_JPEG2000      = /opt/kma/kma_lib/apps/openjpeg/2.3.1/include/
LIB_JPEG2000      = /opt/kma/kma_lib/apps/openjpeg/2.3.1/lib/
INC_ECCODES     = /opt/kma/kma_lib/apps/eccodes/2.16.0/INTEL/210/include/
LIB_ECCODES     = /opt/kma/kma_lib/apps/eccodes/2.16.0/INTEL/210/lib/
INC_NETCDF      = /opt/kma/kma_lib/apps/netcdf/4.7.3/INTEL/210/include/
LIB_NETCDF      = /opt/kma/kma_lib/apps/netcdf/4.7.3/INTEL/210/lib/
INC_HDF4        = /opt/kma/kma_lib/apps/hdf4/4.2.15_fort/INTEL/210/include/
LIB_HDF4        = /opt/kma/kma_lib/apps/hdf4/4.2.15_fort/INTEL/210/lib/
INC_HDF5        = /opt/kma/kma_lib/apps/hdf5/1.10.3/INTEL/210/include/
LIB_HDF5        = /opt/kma/kma_lib/apps/hdf5/1.10.3/INTEL/210/lib/
INC_HDFEOS      = /opt/kma/kma_lib/apps/hdfeos2/2.20/INTEL/210/include/
LIB_HDFEOS      = /opt/kma/kma_lib/apps/hdfeos2/2.20/INTEL/210/lib/
INC_FORTRANGIS1 = 
INC_FORTRANGIS2 = 
LIB_FORTRANGIS  = 
INC_LIBGEOTIFF  = 
LIB_LIBGEOTIFF  = 
LIB_GDAL        = 
CFLAGS          = -c  -DIFC
FFLAGS77        = -c -O2 -nomixed-str-len-arg -names lowercase -convert big_endian -assume byterecl  -DSPMD  -DIFC -I$(MOD_ESMF) -DUSE_INCLUDE_MPI -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5) 
FFLAGS          =  -c -O2 -u -traceback -fpe0  -nomixed-str-len-arg -names lowercase -convert big_endian -assume byterecl  -DSPMD -DIFC -I$(MOD_ESMF) -DUSE_INCLUDE_MPI -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5) 
LDFLAGS         =  -L$(LIB_ESMF) -lesmf -lstdc++ -limf -lm -lrt -lz -lmpi -L$(LIB_ECCODES) -leccodes_f90 -leccodes -L$(LIB_JPEG2000) -lopenjp2 -L$(LIB_NETCDF) -lnetcdff -lnetcdf -L$(LIB_HDFEOS) -lhdfeos -lGctp -L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz  -L$(LIB_HDF5) -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl

Thanks a lot!

**Comment by bmcandr** [comment, marked as answer]
Hi @jachungithub, a quick Google search suggests that your HDF4 library may have been built with SZIP support and needs to be linked when you compile LISF components.

Try editing your `configure.ldt` file to add `-lsz` after `-L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz` on the last line (`LDFLAGS = ...`). You may also need to add the path to the SZIP installation to your `LD_LIBRARY_PATH` environment variable.

See these links for descriptions of this error and potential solutions:
* https://www.ncl.ucar.edu/Support/talk_archives/all/1880.html
* https://netcdfgroup.unidata.ucar.narkive.com/1oRjmi1y/missing-lm-and-lsz-library-issue#post5

**Reply by jachungithub** [reply]
Thanks a lot!
Finally, the compiling of LDT/LIS has been finished with a warning message:
> ld: warning: libjpeg.so.7, needed by /opt/kma/kma_lib/apps/netcdf-fortran/4.5.2/INTEL/210/lib/libnetcdff.so, may conflict with libjpeg.so.9

However, I expect that no more issues will be coming up at LDT/LIS simulations.
In case, any suggestions will be welcome on the warning message above.

Thanks again,
